simpleviewinc/keg-hub
# Zen 526 upgrade expo sdk

**Ticket**: [ZEN-526](https://jira.simpleviewtools.com/browse/ZEN-526)

> NOTES:
> This is the second part of the expo updates.
> These updates do include the expo-sdk changes.
> If you have not do so, please review the changes in this [PR first](https://github.com/simpleviewinc/keg-hub/pull/144)
> This should be tested along side this [PR for keg-cli](https://github.com/simpleviewinc/keg-cli/pull/44)

## Context
* Update Node and Expo CLI/SDK

## Goal
* Update dependencies, and ensure everything still works

## Updates
* Rollup updates
  * Updated to latest rollup for `jsutils` / `keg-components` / `re-theme`
  * Updated rollup plugins to the latest versions
  * Minor updates to the `rollup.config.js` files to work with latest plugins
* Updated `babelHelpers` to be included in the bundle
  * I'm still not crazy about this, but I don't really know a way around it
  * We may want to do a build without the helpers
  * Then include the `@babel/runtime` in keg-core
  * That could make the bundle smaller, but would also add `@babel/runtime` as a dependency
* Updated dependencies across all repos to be up-to-date and consistent
  * Ensured all keg-hub repos use the same version of a dependency
* Fixed `re-theme` build process
  * `re-theme` didn't have index files for the exports so I added them
  * It now matches the `keg-components` build process
* Fixed an issue with `retheme#useStyleTag` on native platforms
  * Now has matching returned objects

## Testing

**Setup**
* Pull this branch => `keg hub && keg pr 145`
* Switch to this [branch](https://github.com/simpleviewinc/keg-cli/pull/44) for the keg-cli
  * Run `keg cli && keg pr 44`

**Test 1**
* Run `install` and `test` commands for each keg-hub repo
  * **Notes** - We should be able to run`keg hub run <command>` for all repos, and have it run in **sync**
    * There's a bug in the `keg-cli`, and instead they are run `async` for all repos
    * I'm looking into it, but for now we have to do them one by one   
  * Repos: `args-parse`, `ask-it`, `jsutils`, `keg-components`, `keg-core`, `spawn-cmd`, `tap-resolver`
  * Command: `keg hub run install <repo-name> && keg hub run test <repo-name>`
    * Replace `<repo-name>` with the repo name from the repos list above
  * Quick copy and paste:
    * **args-parse** - `keg hub run install args-parse && keg hub run test args-parse`
    * **ask-it** - `keg hub run install ask-it && keg hub run test ask-it`
    * **jsutils -** `keg hub run install jsutils && keg hub run test jsutils`
    * **keg-components** - `keg hub run install keg-components && keg hub run test keg-components`
    * **keg-core** - `keg hub run install keg-core && keg hub run test keg-core`
    * **spawn-cmd** - `keg hub run install spawn-cmd && keg hub run test spawn-cmd`
    * **tap-resolver** - `keg hub run install tap-resolver && keg hub run test tap-resolver`
  * Ensure all tests pass from the commands above
* Run `install` and `test` command for ``re-theme`
  * `keg retheme && yarn install && yarn test` 
  * Ensure all tests pass from the commands above

**Test 2**
* Run this keg-components package => `keg dpg run keg-components:zen-526-upgrade-expo-sdk`
  * It this image contains the code from this PR for `jsutils`, `re-theme`, and `keg-component`
  * Ensure you can access storybook in the browser
    * http://components-zen-526-upgrade-expo-sdk.local.kegdev.xyz/
  * Click on the `Ellipsis` icon next to the Storybook logo, then select `About your Storybook`
    * Looks like [this](http://snpy.in/2WZ7oj)
    * Ensure storybook is using version 6
  * Navigate to each component and ensure they render as they did before
  * In another tab attach to the container => `keg components att`
    * Once attached run the following commands
      * `expo -V` => Validates the expo version
        * Should print `4.3.0`
      * `node -v` => Validates the node version
        * Should print `v14.16.0`
  * Before reporting any found issues, please ensure they did not exist prior to this update
    * If they are not related to this update they should be resolved in a separate issue

**Test 3**
* Run this tap-evf package => `keg evf pack run evf:zen-544-def-library`
  * I happened to be on the `zen-544-def-library` branch when I started the tap. 
  * It this image contains the code from this PR for `jsutils`, `re-theme`, and `keg-component`, and `keg-core`
  * Ensure you can access the tap in the browser
    * http://evf-zen-544-def-library.local.kegdev.xyz/
  * Ensure the tap loads, works, and looks as expected
  * In another tab attach to the container => `keg evf att`
    * Once attached run the following commands
      * `expo -V` => Validates the expo version
        * Should print `4.3.0`
      * `node -v` => Validates the node version
        * Should print `v14.16.0`
    * [Looks like this](http://snpy.in/8mUSmg)
  * Before reporting any found issues, please ensure they did not exist prior to this update
    * If they are not related to this update they should be resolved in a separate issue

> Notes
> One cool thing about this is 0 changes are required for the tap, yet it gets all updates
> Image if we had 5,10 or 20 taps
> Without the keg, we would have to go update each tap individually, as well as test and sending PRs
> I rather take a hammer to the face then write the notes for that many PRs
